### ISSUE TITLE
GitLab support

### ISSUE THREAD

**Issue by sanmai-NL** [japaric/rust-everywhere#56]
GitLab support can be had by adding a [`.gitlab-ci.yml`](https://docs.gitlab.com/ce/ci/quick_start/README.html) file.

Release artifacts can be produced easily, see the [GitLab CI reference manual](http://docs.gitlab.com/ce/ci/yaml/README.html#artifacts).

Is any work underway for adding this feature? If desired, I can work on this.

**Comment by japaric** [comment]
> Is any work underway for adding this feature?

No, not really. I'm not familiar with GitLab. (I should check it out sometime :smile:)

> If desired, I can work on this.

That would be awesome. We'd love to support more CI services.

**Comment by sanmai-NL** [comment]
Yeah you should. It has a lot more to offer than GitHub at this point, including native CI.

Okay, great. I'm working on it today for at most an hour. I'll update if I cannot file a PR by Sunday.

**Comment by sanmai-NL** [comment]
Sorry, I doesn't seem to be productive to add a GitLab CI config at this moment. GitLab CI depends on running builds/tests in runners. The default cost-free runners for GitLab.com projects are Linux-based. Travis CI and AppVeyor can spin up macOS and Windows VMs as well. We need support for building for those platforms obviously.

Instead of setting up VMs with the appropriate OSes I prefer to test cross-compilation first in a new purely GitLab-based project similar to this one. In this project you instead use native compilation on macOS/Windows VMs. That is to be able to test as well as build, isn't it?

**Comment by japaric** [comment]
> That is to be able to test as well as build, isn't it?

Yes. It's also much simpler to compile natively than cross compile.

> I prefer to test cross-compilation first in a new purely GitLab-based project

I think you'll only be able to do Linux -> Windows-GNU. Crossing to Windows-MSVC or macOS requires, I think, proprietary libraries which can't be freely distributed.

Still, having binary releases for Linux, perhaps even for different arches: i686, x86_64, arm, etc. via GitLab would be great progress!
